lib/autoproj/cli/main_ci.rb

```diff
@@ -15,22 +31,21 @@ class MainCI < StandaloneCI
             option :report, type: "string", default: "cache-pull.json",
                             desc: "a file which describes what has been pulled"
             def build(*args)
-                if (cache = options[:cache]) || File.exist?(options[:report])
+                if (cache = options[:cache])
+                    Autoproj.warn "ci build: option --cache is deprecated, use "\
+                                  "'ci cache-pull' explicitily instead"
+                end
+
+                report = options[:report]
+                if cache || File.exist?(report)
```

Comment:
I'm having problem with that logic. I get it that it is to handle the default, but then a wrong --report option will be silently ignored.

Maybe use lazy_default instead (from thor docs: `:lazy_default — A default that is only passed if the cli option is passed without a value.`) and throw if the report does not exist ?

Reply:
> but then a wrong --report option will be silently ignored.

Only if `--cache` is not used. I can `raise` if you prefer but I don't think this API won't be ideal until `--cache` is removed.